### PR TITLE
Clean up use of deprecated Mbed OS functions

### PR DESCRIPTION
**Pull request by kjbracey**
Mbed OS 6 will be dropping some deprecated functions, so update code.
No functional change.

### CODE REVIEW

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
cc @ARMmbed/team-st-mcd

**Comment by jeromecoutant** [comment]
@betzw

**Comment by betzw** [comment]
@avilei

**Review by betzw — APPROVED** [review]

File: source/NanostackRfPhySpirit1.cpp [hunk 1]
```diff
@@ -611,17 +611,15 @@ static void rf_ack_loop(void) {
 
     do {
         /* Wait for signal */
-        osEvent event = rf_ack_sender->signal_wait(0);
+        uint32_t signals = ThisThread::flags_wait_any(RF_SIG_ACK_NEEDED | RF_SIG_CB_TX_DONE | RF_SIG_CB_RX_RCVD);
 
-        if(event.status != osEventSignal) {
+        if(!(signals & (RF_SIG_ACK_NEEDED | RF_SIG_CB_TX_DONE | RF_SIG_CB_RX_RCVD)) {
```
Comment (avilei) [inline source/NanostackRfPhySpirit1.cpp:616]: Is this check really necessary considering that the previous line explicitly waits for any of those signals?

Comment (kjbracey) [reply source/NanostackRfPhySpirit1.cpp:616]: It should only occur in the event of some sort of OS error - it's the equivalent of the test it's replacing. I decided to leave it for the "no functional change" claim - you'd still get the trace output for that case.

**Review by avilei — APPROVED** [review]

**Comment by 0xc0170** [comment]
Can anyone merge this PR?

**Pull request merged**